clients/drcachesim/reader/record_file_reader.h

```diff
@@ -149,7 +149,13 @@ class record_reader_t : public std::iterator<std::input_iterator_tag, trace_entr
         UNUSED(res);
         if (!eof_) {
             ++cur_ref_count_;
-            if (type_is_instr(static_cast<trace_type_t>(cur_entry_.type)))
+            // We increment the instr count at the encoding as that avoids multiple
```

Comment:
In a larger app I tracked down a further problem that looks like it should modify this PR rather than being separate, as the ordinals in the current PR here are wrong and it seems best to not merge this as is: we have to account for TRACE_MARKER_TYPE_BRANCH_TARGET which appears between the encoding and the instruction records.  Do we have an existing unit test for record_reader?  Not sure how to make a test of this: will see if it fits in any existing unit test framework.

Reply:
Done, for record_reader_t and scheduler output stream and scheduler switches, with tests added.